### PR TITLE
Fix segfaults

### LINKED ISSUE #265: 0.27.2 segfaults with `invalid memory address or nil pointer dereference`
```
01:30:16 systemd[1]: Started NeoFS HTTP Gateway.
01:30:17 neofs-http-gw[67393]: 2023-08-29T01:30:17.265Z        info        neofs-http-gw/app.go:166        add connection        {"address": "st1.t5.fs.neo.org:8080", "weight": 1, "priority": 2}
01:30:17 neofs-http-gw[67393]: 2023-08-29T01:30:17.265Z        info        neofs-http-gw/app.go:166        add connection        {"address": "st2.t5.fs.neo.org:8080", "weight": 1, "priority": 1}
01:30:17 neofs-http-gw[67393]: 2023-08-29T01:30:17.266Z        info        neofs-http-gw/app.go:166        add connection        {"address": "st3.t5.fs.neo.org:8080", "weight": 1, "priority": 2}
01:30:17 neofs-http-gw[67393]: 2023-08-29T01:30:17.266Z        info        neofs-http-gw/app.go:166        add connection        {"address": "st4.t5.fs.neo.org:8080", "weight": 1, "priority": 2}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.637Z        info        neofs-http-gw/app.go:202        rpc endpoint        {"address": "https://rpc2.morph.t5.fs.neo.org:51331"}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.981Z        info        neofs-http-gw/app.go:320        starting application        {"app_name": "neofs-http-gw", "version": "v0.27.2"}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        neofs-http-gw/app.go:452        added path /upload/{cid}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        neofs-http-gw/app.go:455        added path /get/{cid}/{oid}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        neofs-http-gw/app.go:458        added path /get_by_attribute/{cid}/{attr_key}/{attr_val:*}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        neofs-http-gw/app.go:460        added path /zip/{cid}/{prefix}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        metrics/service.go:33        service hasn't started since it's disabled        {"service": "Pprof"}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.982Z        info        metrics/service.go:27        service is running        {"service": "Prometheus", "endpoint": "localhost:9921"}
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.983Z        info        neofs-http-gw/app.go:490        added server        {"address": "localhost:8888", "tls enabled": false, "tls cert": "", "tls > 
01:30:20 neofs-http-gw[67393]: 2023-08-29T01:30:20.983Z        info        neofs-http-gw/app.go:343        starting server        {"address": "localhost:8888"}
01:41:56 neofs-http-gw[67393]: 2023-08-29T01:41:56.492Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:57978", "method": "GET", "path": "/get_by_attribute/7CpJ>
01:58:54 neofs-http-gw[67393]: 2023-08-29T01:58:54.173Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:47700", "method": "GET", "path": "/get_by_attribute/7CpJ>
02:15:54 neofs-http-gw[67393]: 2023-08-29T02:15:54.189Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:47060", "method": "GET", "path": "/get_by_attribute/7CpJ>
02:36:54 neofs-http-gw[67393]: 2023-08-29T02:36:54.527Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:39812", "method": "GET", "path": "/get_by_attribute/7CpJ>
02:55:54 neofs-http-gw[67393]: 2023-08-29T02:55:54.160Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:36984", "method": "GET", "path": "/get_by_attribute/7CpJ>
03:08:35 neofs-http-gw[67393]: 2023-08-29T03:08:35.626Z        info        neofs-http-gw/app.go:467        request        {"remote": "127.0.0.1:58778", "method": "GET", "path": "/get_by_attribute/7CpJ>
03:08:35 neofs-http-gw[67393]: 2023-08-29T03:08:35.804Z        error        downloader/download.go:367        read object list failed        {"cid": "7CpJVtBdNvPjjdYwV7q9CghGVPagVLTs71BPQuGQLKSQ", "at>
03:08:35 neofs-http-gw[67393]: panic: runtime error: invalid memory address or nil pointer dereference
03:08:35 neofs-http-gw[67393]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0xc1257f]
03:08:35 neofs-http-gw[67393]: goroutine 8952 [running]:
03:08:35 neofs-http-gw[67393]: github.com/nspcc-dev/neofs-http-gw/downloader.(*Downloader).byAttribute(0xc0014973b0, 0xc00184cb00, 0x1089cc8)
03:08:35 neofs-http-gw[67393]:         github.com/nspcc-dev/neofs-http-gw/downloader/download.go:368 +0xdff 
03:08:35 neofs-http-gw[67393]: github.com/nspcc-dev/neofs-http-gw/downloader.(*Downloader).DownloadByAttribute(...)
03:08:35 neofs-http-gw[67393]:         github.com/nspcc-dev/neofs-http-gw/downloader/download.go:327
03:08:35 neofs-http-gw[67393]: main.(*app).logger.func1(0xc00184cb00)
03:08:35 neofs-http-gw[67393]:         github.com/nspcc-dev/neofs-http-gw/app.go:472 +0x4e7 
03:08:35 neofs-http-gw[67393]: github.com/fasthttp/router.(*Router).Handler(0xc0002df570, 0xc00184cb00)
03:08:35 neofs-http-gw[67393]:         github.com/fasthttp/router@v1.4.1/router.go:414 +0x46a 
03:08:35 neofs-http-gw[67393]: github.com/valyala/fasthttp.(*Server).serveConn(0xc0002466c0, {0x11936a8?, 0xc0006b8680})
03:08:35 neofs-http-gw[67393]:         github.com/valyala/fasthttp@v1.34.0/server.go:2341 +0x128e
03:08:35 neofs-http-gw[67393]: github.com/valyala/fasthttp.(*workerPool).workerFunc(0xc001846dc0, 0xc0014883e0)
03:08:35 neofs-http-gw[67393]:         github.com/valyala/fasthttp@v1.34.0/workerpool.go:224 +0xa9
03:08:35 neofs-http-gw[67393]: github.com/valyala/fasthttp.(*workerPool).getCh.func1()
03:08:35 neofs-http-gw[67393]:         github.com/valyala/fasthttp@v1.34.0/workerpool.go:196 +0x38
03:08:35 neofs-http-gw[67393]: created by github.com/valyala/fasthttp.(*workerPool).getCh
03:08:35 neofs-http-gw[67393]:         github.com/valyala/fasthttp@v1.34.0/workerpool.go:195 +0x1b0 
03:08:35 systemd[1]: neofs-http.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
03:08:35 systemd[1]: neofs-http.service: Failed with result 'exit-code'.
03:08:35 systemd[1]: neofs-http.service: Consumed 10.633s CPU time.
03:08:41 systemd[1]: neofs-http.service: Scheduled restart job, restart counter is at 7.
03:08:41 systemd[1]: Stopped NeoFS HTTP Gateway.
03:08:41 systemd[1]: neofs-http.service: Consumed 10.633s CPU time.
03:08:41 systemd[1]: Started NeoFS HTTP Gateway.
```

### PR DESCRIPTION
**Pull request by smallhive**
close #265

### CODE REVIEW

File: downloader/download.go [hunk 1]
```diff
@@ -358,14 +358,19 @@ func (d *Downloader) byAttribute(c *fasthttp.RequestCtx, f func(request, *pool.P
 	if n == 0 {
 		err = res.Close()
 
-		if errors.Is(err, io.EOF) {
-			log.Error("object not found", zap.Error(err))
-			response.Error(c, "object not found", fasthttp.StatusNotFound)
+		if err != nil {
+			if errors.Is(err, io.EOF) {
```
Comment (roman-khimov) [inline downloader/download.go:362]: `if err == nil || errors.Is(err, io.EOF)`

File: downloader/download.go [hunk 2]
```diff
@@ -358,14 +358,19 @@ func (d *Downloader) byAttribute(c *fasthttp.RequestCtx, f func(request, *pool.P
...
+				log.Error("object not found", zap.Error(err))
+				response.Error(c, "object not found", fasthttp.StatusNotFound)
+				return
+			}
+
+			log.Error("read object list failed", zap.Error(err))
```
Comment (roman-khimov) [inline downloader/download.go:368]: All the other cases are this, some errors which are not EOF.

**Review by roman-khimov — COMMENTED** [review]
` tests: Fix panic, if object not found ` --- not `tests`

Code can be simplified a little, but OK as a fix in general.

File: downloader/download.go (hunk 1 above)
Comment (smallhive) [reply downloader/download.go]: Updated

File: downloader/download.go (hunk 2 above)
Comment (smallhive) [reply downloader/download.go]: Updated

**Review by roman-khimov — APPROVED** [review]

**Pull request merged**